src/components/PreviewDraft/PreviewVersion.tsx

```diff
@@ -0,0 +1,162 @@
+  }
+
+  .diffins {
+    background-color: ${colors.support.greenLight};
+    text-decoration: none;
+    display: inline-block;
+  }
+
+  del.diffdel {
+    background-color: ${colors.support.redLight};
+    text-decoration: none;
+    display: inline-block;
+  }
+`;
+
+export const PreviewVersion = ({ article, language, customTitle }: VersionPreviewProps) => {
+  const [diffEnable, setDiffEnable] = useState(true);
+  const { t } = useTranslation();
+  const { values, initialValues } = useFormikContext<LearningResourceFormType>();
+  const { data: licenses = [] } = useLicenses();
+  const apiType = useMemo(
+    () => learningResourceFormTypeToDraftApiType(values, initialValues, licenses),
+    [initialValues, licenses, values],
+  );
+
+  const formArticle: FormArticle = {
+    id: article.id,
+    articleType: article.articleType,
+    title: apiType.title ?? "",
+    content: apiType.content ?? "",
+    introduction: apiType.introduction ?? "",
+    visualElement: apiType.visualElement,
+    published: apiType.published,
+    copyright: apiType.copyright,
+  };
+
+  const publishedArticle = toFormArticle(article, language);
+  const publishedTransformed = useTransformedArticle({ draft: publishedArticle, language });
+  const currentTransformed = useTransformedArticle({ draft: formArticle, language });
+  const currentObj = useMemo(() => {
+    if (!diffEnable) return null;
+    return {
+      title: renderToString(currentTransformed.article?.title as ReactElement),
+      introduction: renderToString(currentTransformed.article?.introduction as ReactElement),
+      content: renderToString(currentTransformed.article?.content as ReactElement),
+    };
+  }, [diffEnable, currentTransformed]);
+  const publishObj = useMemo(() => {
+    if (!diffEnable) return null;
+    return {
+      title: renderToString(publishedTransformed.article?.title as ReactElement),
+      introduction: renderToString(publishedTransformed.article?.introduction as ReactElement),
+      content: renderToString(publishedTransformed.article?.content as ReactElement),
+    };
+  }, [diffEnable, publishedTransformed]);
+
+  const diffObj = useMemo(() => {
+    if (publishObj && currentObj) {
+      const titleDiff = getDiff(publishObj.title, currentObj.title);
+      const introductionDiff = getDiff(publishObj.introduction, currentObj.introduction);
+      const contentDiff = getDiff(publishObj.content, currentObj.content);
+      return { titleDiff, introductionDiff, contentDiff };
+    }
+    return null;
+  }, [currentObj, publishObj]);
+
+  const changeDiff = useCallback(() => {
+    setDiffEnable((p) => !p);
+  }, [setDiffEnable]);
+
+  if (diffEnable && diffObj) {
+    if (publishedTransformed.article && currentTransformed.article) {
+      currentTransformed.article.title = parse(diffObj.titleDiff);
+      currentTransformed.article.introduction = parse(diffObj.introductionDiff);
+      currentTransformed.article.content = parse(diffObj.contentDiff);
```

Comment:
Nå spør jeg kanskje dumt, men er det ikke en viss fare for at dette kan trigge `currentObj` igjen og til slutt føre til at dette renderer veldig mange ganger?

Reply:
Aner ikke, for å være ærlig. Men jeg kan endre det :+1: